src/mlpack/core/data/image_layout_impl.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * @file core/data/image_layout_impl.hpp
+ * @author Andrew Furey
+ *
+ * Convert image layouts between mlpack and stb.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_LAYOUT_IMPL_HPP
+#define MLPACK_CORE_DATA_IMAGE_LAYOUT_IMPL_HPP
+
+#include "image_layout.hpp"
+#include "image_info.hpp"
+
+namespace mlpack {
+namespace data {
+
+template <typename eT>
+arma::Mat<eT> ImageLayout(const arma::Mat<eT>& image,
+    const mlpack::data::ImageInfo& info)
+{
+  arma::Mat<eT> output =
+    arma::vectorise(
+      arma::reshape(image, info.Channels(), info.Height() * info.Width()).t()
+    );
```

Comment:
I believe this only works for one image at a time, right?  So if `image` had multiple columns (e.g. multiple images), then this would actually do something incorrect.  I think it would crash since the `reshape()` size wouldn't be right.

You might consider instead something like this:

```
for (size_t i = 0; i < image.n_cols; ++i)
{
  arma::Mat<eT> alias;
  MakeAlias(alias, image, info.Channels(), info.Height() * info.Width(), i * info.n_rows /* offset by the right number of elements */);
  // then do the .t() into output, maybe make an alias of output too once you set its size
}
```

But I think the logic is right; this is a pretty nice implementation, it's way simpler than the last one.